mono/metadata/profiler.c

```diff
@@ -743,6 +789,38 @@ mono_profiler_class_loaded (MonoClass *klass, int result)
 	}
 }
 
+void
+mono_profiler_static_field (MonoClassField *field)
+{
+	for (ProfilerDesc *prof = prof_list; prof; prof = prof->next)
+		if ((prof->events & MONO_PROFILE_FIELD_EVENTS) && prof->static_field)
+			prof->static_field (prof->profiler, field);
+}
+
+void
+mono_profiler_static_fields_allocated (MonoClass *klass)
```

Comment:
Well, there we go, we have the layout event I mentioned above. Except this one is not properly named.

The static fields didn't get allocated, they layout'd. Otherwise this callback is obviously missing a pointer to the actual memory region.